File: app/src/main/java/com/femtioprocent/omega/anim/appl/ToolBar_Base.java
package com.femtioprocent.omega.anim.appl;

import com.femtioprocent.omega.swing.ToolBar;

import java.util.HashMap;

public abstract class ToolBar_Base extends ToolBar {
    ToolBar_Base(Object o) {
        super();
        init(o);
    }

    ToolBar_Base(Object o, int orientation) {
        super(orientation);
        init(o);
    }

    abstract protected void init(Object o);

    public void populate() {
        populate("default");
    }

    public void populate(String id) {
    }

    public void enable_path(int mask) {
    }
}

File: app/src/main/kotlin/com/femtioprocent/omega/anim/appl/ToolBar_Base.kt
package com.femtioprocent.omega.anim.appl

import com.femtioprocent.omega.swing.ToolBar

abstract class ToolBar_Base : ToolBar {
    internal constructor(o: Any?) : super() {
	init(o)
    }

    internal constructor(o: Any?, orientation: Int) : super(orientation) {
	init(o)
    }

    protected abstract fun init(o: Any?)

    fun populate(id: String? = "default") {
    }

    open fun enable_path(mask: Int) {}
}
